core/node/rpc/syncv3/service.go

```diff
@@ -0,0 +1,104 @@
+package syncv3
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/trace"
+
+	. "github.com/towns-protocol/towns/core/node/base"
+	. "github.com/towns-protocol/towns/core/node/protocol"
+	"github.com/towns-protocol/towns/core/node/rpc/syncv3/handler"
+	. "github.com/towns-protocol/towns/core/node/shared"
+)
+
+// Service defines the behavior of the sync V3 service.
+type Service interface {
+	// SyncStreams creates and starts a sync operation. Given streams are immediately going to be added
+	// to the sync operation, and the receiver will receive updates for these streams.
+	SyncStreams(ctx context.Context, id string, streams []*SyncCookie, rec handler.Receiver) error
```

Comment:
Is it possible to be more precise? In this case it is guaranteed that the caller always receives >= 1 `SyncStreamsResponse` when this function returns nil. It would be nice to desribe these guarantees and expectations concrete.